### PR TITLE
feat: support msvc

### PR DESCRIPTION
**Pull request by zeromake**
dep on [#1](https://github.com/krkrsdl2/krkrz/pull/1)

### CODE REVIEW

File: src/core/base/sdl2/StorageImpl.cpp [hunk 1]
```diff
@@ -289,7 +289,8 @@ void TJS_INTF_METHOD tTVPFileMedia::GetLocallyAccessibleName(ttstr &name)
 			newname = TJS_W("");
 		} else {
 			tjs_char dch = *ptr;
-			if(*ptr < TJS_W('a') || *ptr > TJS_W('z')) {
+			// dch not in 'a'-'z', 'A'-'Z' 
+			if (!(dch >= TJS_W('a') && dch <= TJS_W('z')) && !(dch >= TJS_W('A') && dch <= TJS_W('Z'))) {
```
Comment (zeromake) [inline src/core/base/sdl2/StorageImpl.cpp:293]: support `A-Z` path

File: src/core/base/sdl2/SysInitImpl.cpp [hunk 2]
```diff
@@ -926,7 +928,7 @@ void TVPInitializeBaseSystems()
 
 	// set default current directory
 	{
-		TVPSetCurrentDirectory( IncludeTrailingForwardslash(ExtractFileDir(ExePath())) );
+		TVPSetCurrentDirectory(IncludeTrailingSlash(ExtractFileDir(ExePath())));
```
Comment (zeromake) [inline src/core/base/sdl2/SysInitImpl.cpp:931]: Prevent it from appearing `\/`

File: src/core/msg/sdl2/MsgImpl.cpp [hunk 3]
```diff
@@ -102,1264 +103,8 @@ ttstr TVPReadAboutStringFromResource() {
 	}
 	return ret;
 #endif
-	const char *buf = NULL;
+	const char *buf = __license;
```
Comment (zeromake) [inline src/core/msg/sdl2/MsgImpl.cpp:106]: msvc not support too long strings

File: src/core/sound/sdl2/FAudioDevice.cpp [hunk 4]
```diff
@@ -410,20 +410,21 @@ class FAudioStream : public iTVPAudioStream
...
+			FAudioVoice_SetChannelVolumes(SourceVoiceObj, count, channels, FAUDIO_COMMIT_NOW);
 		}
 		else if (count == 1)
 		{
 			FAudioVoice_SetVolume(SourceVoiceObj, CalcVolume(), FAUDIO_COMMIT_NOW);
 		}
 		else
 		{
-			float channels[count];
+			std::vector<float> channels;
```
Comment (zeromake) [inline src/core/sound/sdl2/FAudioDevice.cpp:421]: msvc not support dynamic arrays

File: src/core/visual/sdl2/WindowImpl.h [hunk 5]
```diff
@@ -31,8 +31,7 @@ struct tTVPWindowMessage
 	LRESULT Result;  // result
 };
 #pragma pack(pop)
-typedef bool (STDCALL * tTVPWindowMessageReceiver)
-	(void *userdata, tTVPWindowMessage *Message);
+typedef bool (__stdcall * tTVPWindowMessageReceiver) (void *userdata, tTVPWindowMessage *Message);
```
Comment (zeromake) [inline src/core/visual/sdl2/WindowImpl.h:34]: `STDCALL` not define use __stdcall

**Comment by zeromake** [comment]
Can you put it [krkrz](https://github.com/krkrsdl2/krkrz) merge to this repo,  this makes it easier to submit PRs

**Comment by uyjulian** [comment]
Thanks!  I applied most of your changes.

The reason I keep the krkrz repository separate is so that I can test if any changes I made are an issue in krkrz or  are also an issue of krkrsdl2.

**Pull request closed without merging**